src/provisioning/provisioner.ts

```diff
@@ -0,0 +1,243 @@
+        const details = connections.map(c => c.getProvisionerDetails?.()).filter(c => !!c) as GetConnectionsResponseItem[];
+        return res.send(details);
+    }
+
+    private getConnection(req: Request<{roomId: string, connectionId: string}>, res: Response<GetConnectionsResponseItem>) {
+        const connection = this.connMan.getConnectionById(req.params.roomId, req.params.connectionId);
+        if (!connection) {
+            throw new ApiError("Connection does not exist", ErrCode.NotFound);
+        }
+        if (!connection.getProvisionerDetails)  {
+            throw new ApiError("Connection type does not support updates", ErrCode.UnsupportedOperation);
+        }
+        return res.send(connection.getProvisionerDetails());
+    }
+
+    private async putConnection(req: Request<{roomId: string, type: string}, unknown, Record<string, unknown>, {userId: string}>, res: Response, next: NextFunction) {
+        // Need to figure out which connections are available
+        try {
+            if (!req.body || typeof req.body !== "object") {
+                throw new ApiError("A JSON body must be provided", ErrCode.BadValue);
+            }
+            const eventId = await this.connMan.provisionConnection(req.params.roomId, req.query.userId, req.params.type, req.body);
+            return res.status(202).send({eventId});
+        } catch (ex) {
+            log.warn(`Failed to create connection for ${req.params.roomId}`, ex);
+            return next(ex);
+        }
+    }
+
+    private async patchConnection(req: Request<{roomId: string, connectionId: string}, unknown, Record<string, unknown>, {userId: string}>, res: Response<GetConnectionsResponseItem>, next: NextFunction) {
+        try {
+            const connection = this.connMan.getConnectionById(req.params.roomId, req.params.connectionId);
+            if (!connection) {
+                return next(new ApiError("Connection does not exist", ErrCode.NotFound));
+            }
+            if (!connection.provisionerUpdateConfig || !connection.getProvisionerDetails)  {
+                return next(new ApiError("Connection type does not support updates", ErrCode.UnsupportedOperation));
+            }
+            await connection.provisionerUpdateConfig(req.query.userId, req.body);
+            res.send(connection.getProvisionerDetails());
+        } catch (ex) {
+            next(ex);
+        }
+    }
+
+    private async deleteConnection(req: Request<{roomId: string, connectionId: string}>, res: Response<{ok: true}>, next: NextFunction) {
+        try {
+            const connection = this.connMan.getConnectionById(req.params.roomId, req.params.connectionId);
+            if (!connection) {
+                return next(new ApiError("Connection does not exist", ErrCode.NotFound));
+            }
+            if (!connection.onRemove) {
+                return next(new ApiError("Connection does not support removal", ErrCode.UnsupportedOperation));
```

Comment:
I believe this is triggering for all integrations. Tested for sure with Jira Projects, but I don't see `onRemove` implemented by anything either.